extension/src/openvic-extension/utility/UITools.cpp

```diff
@@ -217,9 +277,15 @@ static bool generate_icon(generate_gui_args_t&& args) {
 static bool generate_button(generate_gui_args_t&& args) {
 	GUI::Button const& button = static_cast<GUI::Button const&>(args.element);
 
-	// TODO - shortcut, clicksound, rotation (?)
+	// TODO - clicksound, rotation (?)
 	const String button_name = Utilities::std_to_godot_string(button.get_name());
+	const String shortcut_key_name = Utilities::std_to_godot_string(button.get_shortcut());
+	Array event_array = get_events_from_shortcut_key(shortcut_key_name);
 
+	ERR_FAIL_COND_V_MSG(
+		shortcut_key_name.length() != 0 && event_array.size() == 0, false,
+		vformat("Unknown shortcut key '%s' for GUI button %s", shortcut_key_name, button_name)
+	);
```

Comment:
Better to print an error message and set `bool ret` to false if the shortcut isn't generated properly, rather than completely fail to generate the button. We do completely fail below in various cases where we can't generate the sprite, which at the time was all the content the button was made up of, but in retrospect it's also better not to completely fail element generation due to a missing texture.

IMO we should modify this function to allow the `godot::Button` to generate even if there's an issue with the shortcut or texture, and the same for `generate_checkbox`, but these changes don't necessarily have to happen in this PR.

Reply:
Yeah we should fix that later as something don't altogether.